log/log.go

```diff
@@ -39,10 +41,63 @@ func (f levelFlag) Set(level string) error {
 	return nil
 }
 
+func setSyslogFormatter(appname, local string) error {
+	if appname != "" && local != "" {
+		return nil
+	}
+
+	fmter, err := newSyslogger(appname, local, origLogger.Formatter)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error creating syslog formatter: %v\n", err)
+		origLogger.Errorf("can't connect logger to syslogs: %v", err)
+		return err
+	}
+	origLogger.Formatter = fmter
+	return nil
+}
+
+func setJSONFormatter() {
+	origLogger.Formatter = &logrus.JSONFormatter{}
+}
+
+type formatLog struct{}
+
+// String implements flag.Value.
+func (f formatLog) String() string {
+	return origLogger.Level.String()
+}
+
+// Set implements flag.Value.
+func (f formatLog) Set(format string) error {
+	u, err := url.Parse(format)
+	if err != nil {
+		return err
+	}
+	if u.Scheme != "logger" {
+		return fmt.Errorf("invalid scheme %s", u.Scheme)
+	}
+	jsonq := u.Query().Get("json")
+	if jsonq == "true" {
+		setJSONFormatter()
+	}
+
+	switch u.Opaque {
+	case "syslog":
+		appname := u.Query().Get("appname")
+		local := u.Query().Get("local")
+		return setSyslogFormatter(appname, local)
+	case "stdout":
+	default:
+		return fmt.Errorf("unsupported logger %s", u.Opaque)
+	}
+	return nil
+}
+
 func init() {
-	// In order for this flag to take effect, the user of the package must call
+	// In order for these flags to take effect, the user of the package must call
 	// flag.Parse() before logging anything.
 	flag.Var(levelFlag{}, "log.level", "Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal].")
+	flag.Var(formatLog{}, "log.format", "If set use a syslog logger or JSON logging. Example: logger:syslog?appname=bob&local=7 or logger:stdout?json=true ")
```

Comment:
Naming consistency: if we name the above `levelFlag`, then we should also name this one `logFormatFlag` instead of `formatLog`.